### ISSUE TITLE
[Bug] Unable to Load Pronunciation of words which have space 

### ISSUE THREAD

**Issue by oyeraghib** [ankidroid/Anki-Android#11365]
###### Reproduction Steps

1.  Tap on "Add Note" from FAB
2.  Tap the attachment icon at the end of field 
3.  Choose "Advanced editor"
4.  Write down a word ( for example "hello" ) and add one space after that ( "hello ")
5.  Choose PRONUNCIATION and then LOAD

###### Expected Result
It should show the user the loaded pronunciation of the word entered.

###### Actual Result
It says "Word not found" as  a Toast message.

###### Few Approaches to solve it :
1. We can use the trim functionality to trim down the extra spaces entered by the user before moving to next screen and it can avoid the potential error.
2.  We can show a better toast message like "Your word contains extra spaces, please remove them to continue!".

###### Debug info
AnkiDroid Version = 2.16alpha62

Android Version = 9

Manufacturer = asus

Model = ASUS_X01BDA

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 9; ASUS_X01BDA Build/PKQ1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/100.0.4896.127 Mobile Safari/537.36

ACRA UUID = e8337654-a2bd-4df5-850a-c4fe10d24fb3

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by oyeraghib** [comment]
I think solution 1 might be more of a better long term solution for that. Please share your views about it or what do you think of any other alternative solutions for it.

**Comment by mikehardy** [comment]
what's the actual search performed out to the pronunciation service? I wonder if it would be better to urlencode, if not done already. If it is done, then perhaps trim yes

**Comment by oyeraghib** [comment]
> what's the actual search performed out to the pronunciation service? I wonder if it would be better to urlencode, if not done already. If it is done, then perhaps trim yes

I am not sure about what is urlencode in AnkiDroid. @mikehardy can you please tell something about it.
Is it something related to this code present in `BeolingusParser.kt` :

```
    private val PRONUNCIATION_PATTERN = Pattern.compile(
        "<a href=\"([^\"]+)\"[^>]*>" +
            "<img src=\"/pics/s1[.]png\"[^>]*title=\"([^\"]+)\"[^>]*>"
    )
```

**Comment by oyeraghib** [comment]
This issue has been fixed in #11752
